regcore/models.py

```diff
@@ -1,21 +1,27 @@
 from django.db import models
 
+from mptt.models import MPTTModel, TreeForeignKey
+
 from regcore.fields import CompressedJSONField
 
 
-class Regulation(models.Model):
+class Regulation(MPTTModel):
+    id = models.TextField(primary_key=True)
+    parent = TreeForeignKey('self', null=True, blank=True, related_name='children', db_index=True)
     version = models.SlugField(max_length=20)
     label_string = models.SlugField(max_length=200)
     text = models.TextField()
     title = models.TextField(blank=True)
     node_type = models.SlugField(max_length=10)
-    children = CompressedJSONField()
     root = models.BooleanField(default=False, db_index=True)
 
     class Meta:
         index_together = (('version', 'label_string'),)
         unique_together = (('version', 'label_string'),)
 
+    class MPTTMeta:
+        order_insertion_by = ['label_string']
```

Comment:
Ack, yes. Maybe it would be simplest to explicitly encode ordering in the ORM--we could `enumerate` the children in the write view, save the index as `position` or something, and use that column for insertion order. Unless there's an adequate key or set of keys already on this model.